android/build.gradle

```diff
@@ -1,17 +1,20 @@
 apply plugin: 'com.android.library'
 
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion safeExtGet('compileSdkVersion', 28)
 
     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 22
+        minSdkVersion safeExtGet('minSdkVersion', 16)
+        targetSdkVersion safeExtGet('targetSdkVersion', 28)
         versionCode 1
         versionName "1.0"
     }
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    compileOnly "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
```

Comment:
Why use `compileOnly` instead `implementation`?

Reply:
good question - I was actually copying the pattern from another module but I'm using a huge number and I chose one of the more popular - react-native-share:

https://github.com/react-native-community/react-native-share/commit/68582c2ce171de51cce6b8fe68009c4a22dee913

react-native-camera uses this style as well:
https://github.com/react-native-community/react-native-camera/blob/master/android/build.gradle

This is not a strong way to make an argument it is the fallacy of "appeal to authority" since other really popular ones are doing it, it must be correct?

Closer to the source (the gradle team) they indicate this:
```
Dependencies whose API is required at compile time but whose implementation is to be provided by a consuming library, application or runtime environment.
```
https://blog.gradle.org/introducing-compile-only-dependencies

There is a counter-post from gradle indicating this style works but it's not the way they see the world, so they recommend implementation:
https://discuss.gradle.org/t/is-it-recommended-to-use-compileonly-over-implementation-if-another-module-use-implementation-already/26699

For my part, I don't care either way as long as it isn't compile :-). But now I know a lot more! compileOnly seems to follow react ecosystem, implementation seems to follow gradle 2018 recommendation. Between the two I guess I'd choose gradle and implementation. Want a change?

Reply:
Thanks for the elaboration! I know a lot more as well :+1: 

I guess there is no "react ecosystem" since the official doc didn't indicate that we should use compileOnly, it's more like a personal or group's understanding to the gradle's doc.

I'm on your side in the conclusion, the RN docs isn't recommend to use it and the gradle team recommend to use `implementation`. We don't have a valid reason to change to `compileOnly`, so from a commit aspect of view, we would keep using `implementation` to not confuse us or the laters :)

Reply:
Ok - I just re-pushed with one clean commit that is now an 'implementation' dependency for react-native